sky/cli.py

```diff
@@ -1706,13 +1717,40 @@ def status(all: bool, refresh: bool, show_spot_jobs: bool, clusters: List[str]):
                           show_all=False,
                           limit_num_jobs_to_show=not all,
                           is_called_by_user=False))
-        click.echo(f'{colorama.Fore.CYAN}{colorama.Style.BRIGHT}Clusters'
-                   f'{colorama.Style.RESET_ALL}')
+        if not ip:
+            click.echo(f'{colorama.Fore.CYAN}{colorama.Style.BRIGHT}Clusters'
+                       f'{colorama.Style.RESET_ALL}')
         query_clusters: Optional[List[str]] = None
         if clusters:
             query_clusters = _get_glob_clusters(clusters)
         cluster_records = core.status(cluster_names=query_clusters,
                                       refresh=refresh)
+        if ip:
+            if len(cluster_records) != 1:
+                with ux_utils.print_exception_no_traceback():
+                    raise RuntimeError(
+                        f'{len(cluster_records)} cluster found. Please specify '
+                        'exactly one cluster (or glob that only matches one '
+                        'cluster) to show its IP address.')
+            cluster_record = cluster_records[0]
+            if cluster_record['status'] != status_lib.ClusterStatus.UP:
+                with ux_utils.print_exception_no_traceback():
+                    raise RuntimeError('Cluster is not in UP status. ')
+            handle = cluster_record['handle']
+            if not isinstance(handle, backends.CloudVmRayResourceHandle):
+                with ux_utils.print_exception_no_traceback():
+                    raise ValueError('Querying IP address is not supported '
+                                     'for local clusters.')
+            head_ip = handle.head_ip
+            if head_ip is None:
+                # Try fetching the IP address again.
+                head_ip = handle.external_ips()[0]
+                if head_ip is None:
+                    with ux_utils.print_exception_no_traceback():
+                        raise RuntimeError('Failed to fetch IP address. '
+                                           'Please try again later.')
```

Comment:
Wondering if there is some better hint we can provide here than just asking to try again later. Brainstorming - what are the reasons IP fetching would fail? Are there any actionable suggestions we can provide to fix those reasons?

Reply:
One possible reason could be network connection issues 🤔or some ray errors

Reply:
I see... I am trying to find a good way to surface those errors for debugging (i.e., print them when `SKYPILOT_DEBUG=1`), but looks like any errors are not propagated through `handle.external_ips()`... any other ideas?